plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsToolkitExplorerFactory.kt

```diff
@@ -54,20 +72,133 @@ class AwsToolkitExplorerFactory : ToolWindowFactory, DumbAware {
         }
 
         val contentManager = toolWindow.contentManager
-        val content = contentManager.factory.createContent(AwsToolkitExplorerToolWindow.getInstance(project), null, false).also {
+
+        val content = contentManager.factory.createContent(ToolkitWebviewPanel.getInstance(project).component, null, false).also {
             it.isCloseable = true
             it.isPinnable = true
         }
         contentManager.addContent(content)
         toolWindow.activate(null)
         contentManager.setSelectedContent(content)
+
+        project.messageBus.connect().subscribe(
+            ToolkitConnectionManagerListener.TOPIC,
+            object : ToolkitConnectionManagerListener {
+                override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                    connectionChanged(project, newConnection, toolWindow)
+                }
+            }
+        )
+
+        project.messageBus.connect().subscribe(
+            AwsConnectionManager.CONNECTION_SETTINGS_STATE_CHANGED,
+            object : ConnectionSettingsStateChangeNotifier {
+                override fun settingsStateChanged(newState: ConnectionState) {
+                    connectionChanged(project, ToolkitConnectionManager.getInstance(project).activeConnection(), toolWindow)
+                }
+            }
+        )
     }
 
     override fun init(toolWindow: ToolWindow) {
         toolWindow.stripeTitle = message("aws.notification.title")
     }
 
+    private fun connectionChanged(project: Project, newConnection: ToolkitConnection?, toolWindow: ToolWindow) {
+        val isToolkitConnected = when (newConnection) {
+            is AwsConnectionManagerConnection -> {
+                getLogger<AwsToolkitExplorerFactory>().debug { "IAM connection" }
+                true
+            }
+
+            is AwsBearerTokenConnection -> {
+                val cond1 = CODECATALYST_SCOPES.all { it in newConnection.scopes }
+                val cond2 = newConnection.scopes.contains(IDENTITY_CENTER_ROLE_ACCESS_SCOPE)
```

Comment:
should have better names

Reply:
I feel we will only need it before 4/30 for bugbash purpose